internal/pkg/flink/internal/controller/interactive_output_controller.go

```diff
@@ -15,36 +15,56 @@ import (
 	"github.com/confluentinc/cli/internal/pkg/output"
 )
 
+const errorDuringTableView = "Error: internal error occurred while in table view"
+
 type InteractiveOutputController struct {
 	app           *tview.Application
 	tableView     components.TableViewInterface
 	resultFetcher types.ResultFetcherInterface
+	reportUsage   func()
 	isRowViewOpen bool
 	debug         bool
 }
 
-func NewInteractiveOutputController(tableView components.TableViewInterface, resultFetcher types.ResultFetcherInterface, debug bool) types.OutputControllerInterface {
+func NewInteractiveOutputController(tableView components.TableViewInterface, resultFetcher types.ResultFetcherInterface, reportUsage func(), debug bool) types.OutputControllerInterface {
 	return &InteractiveOutputController{
 		app:           tview.NewApplication(),
 		tableView:     tableView,
 		resultFetcher: resultFetcher,
+		reportUsage:   reportUsage,
 		debug:         debug,
 	}
 }
 
 func (t *InteractiveOutputController) VisualizeResults() {
+	defer t.panicRecovery()
 	t.init()
-	t.start()
+	t.startTView()
+	t.close()
+}
+
+func (t *InteractiveOutputController) panicRecovery() {
+	if r := recover(); r != nil {
```

Comment:
Surprised that this isn't caught by the main `recover()` function

Reply:
Oh no that does work! It will get caught by the man recover(), it's just that we cannot delete the statement we created there and this can leave dangling resources (because currently there is no cancel API for statements, so we have to delete statements that would otherwise run forever).

So it just gives us a bit more control over how we recover in case of a panic.